tests/rkt_run_pod_manifest_test.go

```diff
@@ -0,0 +1,360 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/ThomasRooney/gexpect"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+)
+
+const baseAppName = "coreos.com/rkt-inspect"
+
+func importImageAndFetchHash(t *testing.T, ctx *rktRunCtx, img string) string {
+	// Import the test image into store manually.
+	cmds := strings.Fields(ctx.cmd())
+	fetchCmd := exec.Command(cmds[0], cmds[1:]...)
+	fetchCmd.Args = append(fetchCmd.Args, "--insecure-skip-verify", "fetch", img)
+	output, err := fetchCmd.Output()
+	if err != nil {
+		t.Fatalf("Cannot read the output: %v", err)
+	}
+
+	// Read out the image hash.
+	ix := strings.Index(string(output), "sha512-")
+	if ix < 0 {
+		t.Fatalf("Unexpected result: %v, expecting a sha512 hash", string(output))
+	}
+	return strings.TrimSpace(string(output)[ix:])
+}
+
+func isEmptyApp(app *types.App) bool {
+	return reflect.DeepEqual(app, &types.App{})
+}
+
+func patchPodManifest(t *testing.T, pod *schema.PodManifest, args ...string) {
+	var flags flag.FlagSet
+	var appName string
+	var imageName string
+	var imageID string
+	var exec string
+	var environments envMap
+	var mountPoints mountPointList
+	var containerPorts containerPortList
+	var exposedPorts exposedPortList
+	var volumes volumeList
+	var user string
+	var group string
+
+	var ra schema.RuntimeApp
+	var app types.App
+
+	flags.StringVar(&appName, "app-name", "", "")
```

Comment:
This feels like we're going around in circles. Why use flags as an in-process interface instead of something more structured?